decoder.go

```diff
@@ -112,14 +112,16 @@ func decodeDiagnosticsEntry(obj *wafObject) (*DiagnosticEntry, error) {
 	return &entry, nil
 }
 
-func decodeDiagnosticAddresses(obj *wafObject) (res DiagnosticAddresses, err error) {
+func decodeDiagnosticAddresses(obj *wafObject) (res *DiagnosticAddresses, err error) {
 	if !obj.isMap() {
 		return res, errInvalidObjectType
 	}
 	if obj.value == 0 && obj.nbEntries > 0 {
 		return res, errNilObjectPtr
 	}
 
+	res = &DiagnosticAddresses{}
```

Comment:
Any chance to return `nil` when no need to have a value?
I also think it would simplify the `switch...case` block when directly returning the values from each `case` block.